### PR TITLE
Fix enum's button value translations

### PR DESCRIPTION
**Pull request by ghost**
Turns out it only worked on a specific edge case, which has having the config name left untranslated, making the change pointless, this PR makes the previous PR that I have made actually functional.

### CODE REVIEW

**Review by shedaniel — APPROVED** [review]

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java [hunk 1]
```diff
@@ -276,19 +276,18 @@ private void initDefaultFunctionMap() {
             @SuppressWarnings("unchecked") ConfigLeaf<String> leaf = ((ConfigLeaf<String>) node);
             EnumSerializableType type = (EnumSerializableType) leaf.getConfigType();
             ClothSetting.EnumHandler.EnumDisplayOption displayOption = leaf.getAttributeValue(ClothAttributes.SUGGESTION_ENUM, ENUM_DISPLAY_TYPE).orElse(ClothSetting.EnumHandler.EnumDisplayOption.BUTTON);
-            String key = getFieldNameKey(leaf.getName()).getString();
+            Text key = new LiteralText("config." + modId + "." + leaf.getName());
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: What's the point in using Text?

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java [hunk 2]
```diff
@@ -276,19 +276,18 @@ private void initDefaultFunctionMap() {
...
-                        .setNameProvider((name) -> {
-                            if (I18n.hasTranslation(key + ".enum." + name.toLowerCase(Locale.ROOT)))
-                                return new TranslatableText(key + ".enum." + name.toLowerCase(Locale.ROOT));
+                        .setNameProvider(name -> {
+                            if (I18n.hasTranslation(key.getString() + ".enum." + name.toLowerCase(Locale.ROOT)))
+                                return new TranslatableText(key.getString() + ".enum." + name.toLowerCase(Locale.ROOT));
                             return new LiteralText(name);
-                        })
-                        .build()
+                        }).build()
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: unrelated change

**Review by shedaniel — CHANGES_REQUESTED** [review]

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java (hunk 1 above)
Comment (ghost) [reply src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: I switched to Text since TranslatableText was causing issues (which led to this PR), however, perhaps I can go back to String since I'm using a LiteralText now

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java [hunk 3]
```diff
@@ -276,14 +276,14 @@ private void initDefaultFunctionMap() {
             @SuppressWarnings("unchecked") ConfigLeaf<String> leaf = ((ConfigLeaf<String>) node);
             EnumSerializableType type = (EnumSerializableType) leaf.getConfigType();
             ClothSetting.EnumHandler.EnumDisplayOption displayOption = leaf.getAttributeValue(ClothAttributes.SUGGESTION_ENUM, ENUM_DISPLAY_TYPE).orElse(ClothSetting.EnumHandler.EnumDisplayOption.BUTTON);
-            String key = getFieldNameKey(leaf.getName()).getString();
+            String key = new LiteralText("config." + modId + "." + leaf.getName()).getString();
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: no, just `"config." + modId + "." + leaf.getName()`

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java [hunk 4]
```diff
@@ -276,14 +276,14 @@ private void initDefaultFunctionMap() {
...
             if (displayOption == ClothSetting.EnumHandler.EnumDisplayOption.BUTTON) {
                 return Collections.singletonList(configEntryBuilder
                         .startSelector(getFieldNameKey(leaf.getName()), type.getValidValues().toArray(new String[0]), leaf.getValue())
                         .setDefaultValue(leaf.getDefaultValue())
                         .setSaveConsumer(leaf::setValue)
                         .setErrorSupplier(v -> error(type, v))
-                        .setNameProvider((name) -> {
+                        .setNameProvider(name -> {
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: again, unrelated change

File: src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java (hunk 3 above)
Comment (ghost) [reply src/main/java/me/shedaniel/fiber2cloth/impl/Fiber2ClothImpl.java]: 🤦
